Update for spikeinterface deprecation of `from_times_labels`

zm711 opened the pull request:
Howdy Jeremy,

we have released a new version of spikeinterface that deprecates `from_times_labels` because it required samples rather than times and so is a mess. There are two solutions for MS5--either set a new floor for si version or this PR where we add the packaging dependency but this lets us use the old function for old version of si and the correctly named function for new versions.

I would recommend packaging if you decide to support si < 0.102.2 because the "dev" tag is no longer used by library so parsing old versions vs new versions will be a mess if we don't use something that handles versioning in python :)

zm711 committed:
update for spikeinterface deprecation

zm711 commented:
Let me know how you want me to fix the indentation and the errors in the integration checks? Are there linter settings I can use?

magland commented:
Thanks @zm711 for this PR!

I'd prefer not to use pyright linting any more since it has been causing problems. Wondering if you would be able to just remove that check from the gh actions files?

zm711 committed:
update actions

zm711 commented:
So I updated actions. Now we can focus on what the test failure was. I also fixed my build action cron job because I just realized I never saved it as a yml so it wasn't being run....

zm711 committed:
add ds_store to gitignore

zm711 commented:
I need to also remove the .DS_Store (it's a Mac folder thing) and just to prevent your repo history from getting messed up with a gross file you'll probably want to squash this if you don't normally squash PRs.

zm711 committed:
remove random .DS_store file

zm711 committed:
remove pyright from integration tests

zm711 committed:
Merge branch 'fix-from_time_labels' of https://github.com/zm711/mountainsort5 into fix-from_time_labels

zm711 commented:
Okay so tests are fixed now, but there is a codecov error with uploading the results. I doubt I would be able to fix this without having write permission to the repo. Did you want to try to rerun the tests once? And if they still don't work then I could also move the codecov to a cron job instead and then you could tweak it later to ensure it uploads correctly.

magland commented:
Looks like the tests ran successfully so I'll merge. Maybe the codecov error will resolve itself.

magland commented:
released new version to pypi
